dicodile/update_z/distributed_sparse_encoder.py

```diff
@@ -67,9 +77,10 @@ def init_workers(self, X, D_hat, reg, params, z0=None, DtD=None):
         )
 
     def set_worker_D(self, D, DtD=None):
-        msg = "The support of the dictionary cannot be changed on an encoder."
-        assert D.shape[1:] == self.D_shape[1:], msg
-        self.D_shape = D.shape
+        msg = "Cannot change dictionary support on an encoder."
+        if not _is_rank1(D):
+            assert D.shape[1:] == self.D_shape[1:], msg
+            self.D_shape = D.shape  # XXX in case number of atoms change?
```

Comment:
Need the second part of this for `rank1` dictionary no?

Reply:
yes, do you think that warrants an additional unit test?

Reply:
No I think it is fine that way.
I would say let's merge this one and move on.

A nice think would be to make an example with rank1 from `alphacsc` to make sure everything works properly?

Reply:
OK, I'll finish addressing your comments in this PR and we can merge.

Agree with the need for an example in alphacsc, I'll start a PR there after addressing the comments here?

Reply:
Yes great!